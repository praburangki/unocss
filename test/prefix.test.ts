import { createGenerator } from '@unocss/core'
import { describe, expect, test } from 'vitest'
import presetUno from '@unocss/preset-uno'

describe('prefix', () => {
  test('preset prefix', async () => {
    const uno = createGenerator({
      presets: [
        presetUno({ prefix: 'h-' }),
      ],
      rules: [
        ['bar', { color: 'bar' }, { prefix: 'bar-' }],
      ],
      shortcuts: [
        ['shortcut', 'bar-bar', { prefix: 'bar-' }],
      ],
    })

    const unexpected = [
      'text-red',
      'hover:p4',
      'bar',
      'shortcut',
    ]

    const expected = [
      'h-text-red',
      'hover:h-p4',
      'bar-bar',
      'bar-shortcut',
      'h-container',
      '2xl:h-container',
    ]

    const { css, matched } = await uno.generate(new Set([
      ...unexpected,
      ...expected,
    ]), { preflights: false })

    expect([...matched].sort()).toEqual(expected.sort())
    expect(css).toMatchSnapshot()
  })

  test('multiple preset prefix', async () => {
    const uno = createGenerator({
      presets: [
        presetUno({ prefix: ['h-', 'x-'] }),
      ],
      rules: [
        ['bar', { color: 'bar' }, { prefix: ['bar-', 'foo-', ''] }],
        [/^regex$/, () => ({ color: 'regex' }), { prefix: ['bar-', 'foo-', ''] }],
      ],
      shortcuts: [
        ['shortcut', 'bar-bar', { prefix: ['bar-', 'x-'] }],
      ],
    })

    const unexpected = [
      'text-red',
      'hover:p4',
      'bar',
      'shortcut',
    ]

    const expected = [
      'h-text-red',
      'hover:h-p4',

      'bar-bar',
      'foo-bar',
      'bar',

      'bar-regex',
      'foo-regex',
      'regex',

      'bar-shortcut',
      'x-shortcut',

      'h-container',
      '2xl:h-container',
    ]

    const { css, matched } = await uno.generate(new Set([
      ...unexpected,
      ...expected,
    ]), { preflights: false })

    expect([...matched].sort()).toEqual(expected.sort())
    expect(css).toMatchSnapshot()
  })
})
