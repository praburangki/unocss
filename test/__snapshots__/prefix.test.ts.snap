// Vitest Snapshot v1

exports[`prefix > multiple preset prefix 1`] = `
"/* layer: default */
.h-container{max-width:100%;}
.hover\\\\:h-p4:hover{padding:1rem;}
.h-text-red{--un-text-opacity:1;color:rgba(248,113,113,var(--un-text-opacity));}
.bar,
.bar-bar,
.bar-shortcut,
.foo-bar,
.x-shortcut{color:bar;}
.bar-regex,
.foo-regex,
.regex{color:regex;}
@media (min-width: 640px){
.h-container{max-width:640px;}
}
@media (min-width: 768px){
.h-container{max-width:768px;}
}
@media (min-width: 1024px){
.h-container{max-width:1024px;}
}
@media (min-width: 1280px){
.h-container{max-width:1280px;}
}
@media (min-width: 1024px){@media (min-width: 1536px){
.\\\\32 xl\\\\:h-container{max-width:1024px;}
}}
@media (min-width: 1280px){@media (min-width: 1536px){
.\\\\32 xl\\\\:h-container{max-width:1280px;}
}}
@media (min-width: 1536px){
.\\\\32 xl\\\\:h-container{max-width:100%;}
.h-container{max-width:1536px;}
}
@media (min-width: 1536px){@media (min-width: 1536px){
.\\\\32 xl\\\\:h-container{max-width:1536px;}
}}
@media (min-width: 640px){@media (min-width: 1536px){
.\\\\32 xl\\\\:h-container{max-width:640px;}
}}
@media (min-width: 768px){@media (min-width: 1536px){
.\\\\32 xl\\\\:h-container{max-width:768px;}
}}"
`;

exports[`prefix > preset prefix 1`] = `
"/* layer: default */
.h-container{max-width:100%;}
.hover\\\\:h-p4:hover{padding:1rem;}
.h-text-red{--un-text-opacity:1;color:rgba(248,113,113,var(--un-text-opacity));}
.bar-bar,
.bar-shortcut{color:bar;}
@media (min-width: 640px){
.h-container{max-width:640px;}
}
@media (min-width: 768px){
.h-container{max-width:768px;}
}
@media (min-width: 1024px){
.h-container{max-width:1024px;}
}
@media (min-width: 1280px){
.h-container{max-width:1280px;}
}
@media (min-width: 1024px){@media (min-width: 1536px){
.\\\\32 xl\\\\:h-container{max-width:1024px;}
}}
@media (min-width: 1280px){@media (min-width: 1536px){
.\\\\32 xl\\\\:h-container{max-width:1280px;}
}}
@media (min-width: 1536px){
.\\\\32 xl\\\\:h-container{max-width:100%;}
.h-container{max-width:1536px;}
}
@media (min-width: 1536px){@media (min-width: 1536px){
.\\\\32 xl\\\\:h-container{max-width:1536px;}
}}
@media (min-width: 640px){@media (min-width: 1536px){
.\\\\32 xl\\\\:h-container{max-width:640px;}
}}
@media (min-width: 768px){@media (min-width: 1536px){
.\\\\32 xl\\\\:h-container{max-width:768px;}
}}"
`;
